src/index.js

```diff
@@ -213,6 +223,37 @@ export default options => {
     }
   }
 
-
-  return { get, add, del, clients, broadcast, empty, clear, transform }
+  /**
+   * Create pruner.
+   * @param {String} id - spark ID
+   * @api private
+   */
+  function prune(id, roomName) {
+    try {
+      let socketsSet = roomsMap.get(roomName);
+      // remove the socket from the socketsInRoom set
+      if (socketsSet instanceof Set) {
+        socketsSet.delete(id);
+        // if no sockets connect to a room remove the room
+        if (socketsSet.size === 0) {
+          roomsMap.delete(roomName);
+          // TODO add wild.remove and update its logic
+        }
+      }
+      let roomsSet = socketsMap.get(id);
+      // remove room from the socket's rooms
+      if (roomsSet instanceof Set) {
+        roomsSet.delete(roomName);
+        // if socket is not connected to any rooms remove the rooms set for it
+        if (roomsSet.size === 0) {
+          socketsMap.delete(id);
+        }
+      }
+      return Promise.resolve({success: true});
+    } catch (e) {
+      return Promise.reject(e);
+    }
+  }
+  
+  return { getRooms, add, del, clients, broadcast, empty }
```

Comment:
Yes please change it back, I prefer short names over long/verbose names, the same apply for the `Set` variable names and any additional variable.